internal/client.go

```diff
@@ -431,50 +432,69 @@ func (c *Client) OnRequestCompleted(rc RequestCompletionCallback) {
 	c.onRequestCompleted = rc
 }
 
+func (c *Client) clearCookies() {
+	c.cookies = nil
+}
+
 // Do performs request passed
 func (c *Client) Do(ctx context.Context, req *http.Request, v interface{}) error {
 	// check if httpClient exists or not
+	maxRetries := 1
```

Comment:
how did we decide the maxRetries as 1?

Reply:
It's really just a retry once. If your refresh cookies and retry and it still fails it should give up and fail rather than keep retrying.